### PR TITLE
build(plop): add plop for components

### PR DESCRIPTION
**Pull request by aaronmaturen**

### CODE REVIEW

File: plop-templates/component-pkg/{{capitalize componentName}}.tsx.hbs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    {{/if}}
+    {{#if useI18n}}
+    const i18n = React.useContext(I18nContext);
+    {{/if}}
+
+    return ({{#if useInverse}}<InverseContext.Provider value=\{{ isInverse, }}>{{/if}}
+      <Styled{{capitalize componentName}}
+      {{#if useTheme}} theme={theme} {{/if}}
+      {{#if useInverse}} isInverse={isInverse}{{/if}}
+      ref={ref}
```
Comment (joasmi723) [inline plop-templates/component-pkg/{{capitalize componentName}}.tsx.hbs:66]: Can we pull the `other` props off the props object and spread them here, like we do in other components?

**Review by joasmi723 — COMMENTED** [review]
This looks awesome!

File: plop-templates/website/component.mdx.hbs [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---
+pageTitle: {{capitalize componentName}} API
+title: {{capitalize componentName}}
+props:
+- {{capitalize componentName}}Props
+---
+
+<DocsHeading>
```
Comment (joasmi723) [inline plop-templates/website/component.mdx.hbs:8]:
```suggestion
<DocsHeading type="api">
```

File: plop-templates/component-pkg/index.ts.hbs [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// export * from "./use{{capitalize componentName}}"
```
Comment (joasmi723) [inline plop-templates/component-pkg/index.ts.hbs:1]: Remove commented line?

Comment (aaronmaturen) [reply plop-templates/component-pkg/index.ts.hbs:1]: I think I want that left there for if we use Hooks... without the line commented we get a typescript error and it's easy to cleanup after it's generated

**Review by joasmi723 — APPROVED** [review]

**Pull request merged**